---
output:
  pdf_document:
    latex_engine: xelatex
header-includes:
    - \usepackage{fancyhdr}
    - \usepackage{fontspec}
    - \usepackage{xcolor}
    - \geometry{left = 0.5in,right = 0.5in}
mainfont: Roboto
sansfont: Roboto
urlcolor: purplepeopleeater
---
<!-- define color and adjust lengths for header and footer-->
\definecolor{purplepeopleeater}{RGB}{106,13,75}
\addtolength{\headheight}{3.0cm}
\addtolength{\topmargin}{-0.5in}
\addtolength{\footskip}{-0.225in}

<!-- % setup header -->
\pagestyle{fancy}
\fancyhf{}

<!-- header content -->
<!-- Uncomment the line of code below to include a header -->
<!-- \fancyhead[L]{\raisebox{-0.25\height}{\includegraphics[height = 2.5cm]{path_to_your_header_here.png}}} -->
\fancyhead[R]{\Huge Genomic Clustering Report\\
\Large `r paste(Sys.Date())`}

<!-- create red header line -->
\renewcommand{\headrulewidth}{1pt}
\renewcommand{\headrule}{\hbox to\headwidth{%
    \color{purplepeopleeater}\leaders\hrule height \headrulewidth\hfill}}

<!-- footer content -->
\fancyfoot[C]{For research use only, not for clinical use.}
\fancyfoot[R]{\thepage}

<!-- create red footer line -->
\renewcommand{\footrulewidth}{1pt}
\renewcommand{\footrule}{\hbox to\headwidth{%
    \color{purplepeopleeater}\leaders\hrule height \headrulewidth\hfill}}

```{r include = FALSE}

### make sure the following libraries are installed
library(ggplot2)
library(ggtree)
library(phytools)

### get date of report generation
date <- Sys.Date()

### set figure size
knitr::opts_chunk$set(out.width = "7.5in", out.height = "8in", fig.align = "left")

### set seed for reproducibility
set.seed(123)

```

### *`r species`* Analysis

<!-- We recommend including the species and number of isolates included in your analysis, as well as a sentence describing the proper use of your results -->

The analysis of the *`r species`* sample (n = `r num.iso`) has been completed.

### SNP Heatmap

[Figure description and results here]

```{r echo = FALSE, message = FALSE, warning = FALSE}

### This block midpoint-roots a tree and plots it with a SNP heatmap

### uncomment this block if you want to manually run this script in Rstudio
### SNP matrix file; change delimiter as necessary
### snp_mat <- read.delim("/path/to/matrix.tsv",
###    header = T,
###    stringsAsFactors = F,
###    sep = "\t")

### uncomment this block if you want to manually run this script in Rstudio
### path to tree in newick format
### nwk <- "/path/to/tree.newick"

### read tree and midpoint root
tree <- read.tree(nwk)
mpt <- midpoint.root(tree)

### uncomment if column names begin with a number
colnames(snp_mat) <- gsub("X","",colnames(snp_mat))

### replace any "." with "-" in column names
colnames(snp_mat) <- gsub("\\.","-",colnames(snp_mat))

### plot midpoint-rooted tree
gtree <- ggtree(mpt, size  = .25, branch.length = "none")

### convert midpoint-rooted tree to dataframe
mpt.fort <- fortify(mpt)

### get vertical order of tip labels
mpt.tip <- mpt.fort[which(mpt.fort$isTip ==  TRUE),]
mpt.ord <- mpt.tip$label[order(mpt.tip$y)]

### order snp matrix by vertical order of tip labels
snp_mat <- snp_mat[c(mpt.ord),c(mpt.ord)]

### alter these plotting defaults as necessary; these are what we've found work best for our data

### heatmap width relative to plot
heatmap_width <- 25

### font size for heatmap row and column names
axis_font_size <- 2.25

### font size for heatmap values
cell_font_size <- 2.25

### tree offset from heatmap
tree_offset <- 18

### offset of column names from heatmap
col_offset <- 0.25

### offset of row names names from heatmap; should be negative
row_offset <- -15

### legend title font size
legend_title_size <- 8

### legend body font size
legend_text_size <- 6

### height of heatmap colourbar
colourbar_height <- 0.5

### width of heatmap colourbar
colourbar_width <- 7

### ggtree will often crop the figure too small; subtract from ymin and add to ymax to fix this
ymin <- min(gtree$data$y) - 4
ymax <- max(gtree$data$y) + 1

### main tree plotting function
gheatmap(gtree, snp_mat,
    width = heatmap_width,
    offset = tree_offset,
    cell_labels = TRUE,
    cell_font_size = cell_font_size,
    font.size = axis_font_size,
    colnames_angle = 90,
    rownames_angle = 0,
    colnames_offset = col_offset,
    rownames_offset = row_offset) +

### set heatmap colourbar colors and limits
scale_fill_viridis_c(limits = c(1,(max(snp_mat)+1)),
    na.value = "white",
    name = "SNPs",
    guide = "colourbar") +

### set plot y limits
ylim(ymin,ymax) +

### remove whitespace around plot and add legend
theme(plot.margin = unit(c(0,0,0,0), "mm"),
    legend.box = "horizontal",
    legend.text = element_text(size = legend_text_size),
    legend.title = element_text(size = legend_title_size),
    legend.position = "bottom",
    legend.margin = margin(0,0,0,0)) +

### place heatmap colourbar beneath the heatmap (rather than beside)
guides(fill = guide_colourbar(title.position = "top",
    title.hjust = 0.5,
    barheight = colourbar_height,
    barwidth = colourbar_width))

```
\newpage

### Phylogenetic tree

[Figure description and results here]

```{r echo = FALSE, message = FALSE, warning = FALSE}

### This block plots the midpint-rooted tree with bootstrap values

### alter these plotting defaults as necessary; these are what we've found work best for our data

### bootstrap cutoff; plot boostrap values above this threshold
boot_thresh <- 95

### size of node label text
node_text_size <- 1.75

### nudge node label text horizontally
x_nudge <- -0.00005

###
scale_offset <- 0.1

### tree scale font size
scale_font_size <- 2

### tip label font size
tip_font_size <- 2

### main tree plotting function
gtree <- ggtree(mpt, color = "black", alpha = 0.75, size = 0.5) +

### add boostrap values as node labels
geom_nodelab(aes(x = branch,
  label = label,
  subset = !isTip & (as.numeric(label) >= boot_thresh)),
  vjust = -0.5,
  nudge_x = x_nudge,
  size = node_text_size) +

### add tip labels
geom_tiplab(size = tip_font_size) +

### add tree scale
geom_treescale(offset = scale_offset,
    fontsize = scale_font_size,
    y = 0,
    x = 0) +

### remove whitespace around plot
theme(plot.margin = unit(c(0,0,0,0), "cm"))

### ggtree will often crop the figure too small; add to xmax to fix this
### we've found the following function calculates a decent value to add to xmax:

log10_ceiling <- function(x) {
    10^(ceiling(log10(x)))
}

xmax <- max(gtree$data$x) + (log10_ceiling(max(gtree$data$x))/5)
xmin <- 0

### set x limits and plot tree
gtree + xlim(xmin,xmax)

```

### Methods

[Methods used to generate data here]
